apps/dav/lib/CalDAV/CalendarImpl.php

```diff
@@ -145,11 +145,16 @@ public function createFromString(string $name, string $calendarData): void {
 		// so set the custom principal here
 		$plugin->setCurrentPrincipal($this->calendar->getPrincipalURI());
 
+		// Build full calendar path
+		$parts = explode('/',  $this->calendar->getPrincipalURI());
```

Comment:
You should use `Sabre\Uri\split` instead.
```suggestion
		[, $user] = Uri\split($this->calendar->getPrincipalURI());
```